File: CPP/test/Test_Steak.cpp
#include "../src/Steak.cpp"
#include <gtest/gtest.h>

using namespace std;

TEST(SteakTest, AllFunctionalityTest){
    Steak testSteak(150);
    LinkedList<FarmProductType> SteakRecipe = testSteak.getResep();
    ASSERT_EQ(MEAT, SteakRecipe.get(0));
    ASSERT_EQ(MILK, SteakRecipe.get(1));
    ASSERT_EQ(MEAT, testSteak.getResepwithIdx(0));
    ASSERT_EQ(MILK, testSteak.getResepwithIdx(1));
    ASSERT_EQ(150, testSteak.getPrice());
    ASSERT_EQ(STEAK, testSteak.getObjectType());
    ASSERT_EQ(SIDEPRODUCT, testSteak.getProductType());
}

File: Java/src/test/java/engisubs/gameobject/product/sideproduct/TestSteak.java
package engisubs.gameobject.product.sideproduct;

import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;

import engisubs.gameobject.GameObject.FarmProductType;
import engisubs.gameobject.GameObject.GameObjectType;
import engisubs.gameobject.product.Product.ProductType;

public class TestSteak{
    private Steak test;

    @Before
    public void setUpObject() {
        test = new Steak(100);
    }
    @Test
    public void instantiationTest(){
        Assert.assertEquals(100, test.getPrice());
        Assert.assertEquals(GameObjectType.STEAK, test.getGameObjectType());
        Assert.assertEquals(ProductType.SIDEPRODUCT, test.getProductType());
        Assert.assertEquals("Medium Rare Espanolia Steak~~", test.getProductName());
    }
    @Test
    public void resepTest(){
        Steak.InitResep();
        Assert.assertArrayEquals(new FarmProductType[]{FarmProductType.MEAT, FarmProductType.EGG}, Steak.getResep().toArray());
        Assert.assertEquals(null, Steak.getResepwithIdx(3));
    }
}